### ISSUE TITLE
Projection should not affect original field name in order by

### ISSUE THREAD

**Issue by StarpTech** [orientechnologies/orientdb#4510]
OrientDb: 2.0.11

When I use projection and change the name of my field from `Created` to `created` then `order by Created` will not work. Aggregations should also work on the original name.

**Comment by nagarajasr** [comment]
+1

**Comment by luigidellaquila** [comment]
+1, it's a nice to have.
It requires some modifications to the query executor, so it cannot be safely implemented at this stage (advanced RC).
I'm moving it to 2.2

**Comment by nagarajasr** [comment]
There is a related bug on where clause:

```sql

orientdb {db=SRIMS}> select out_photos as o from testclass where out_photos is null

----+------
#   |@CLASS
----+------
0   |null
1   |null
2   |null
----+------

3 item(s) found. Query executed in 0.01 sec(s).
orientdb {db=SRIMS}> select out_photos as o from testclass where o is null

----+------+----
#   |@CLASS|o
----+------+----
0   |null  |[3]
1   |null  |null
2   |null  |null
3   |null  |null
----+------+----

4 item(s) found. Query executed in 0.009 sec(s).

```

**Comment by n0lf** [comment]
+1

**Comment by luigidellaquila** [comment]
This is already fixed in 3.0M2

Thanks

Luigi
